catalog/spatial/geocoding/spatial-geocoding-plugin/src/main/java/org/codice/ddf/spatial/geocoding/plugin/GeoCoderPlugin.java

```diff
@@ -118,11 +106,38 @@ public int getRadius() {
         return radiusInKm;
     }
 
-    private boolean countryCodeNeeded(Metacard metacard) {
-        return metacard.getAttribute(Location.COUNTRY_CODE) == null;
+    /**
+     * Sets the country code attribute of the {@param metacard} when the {@param metacard} has a
+     * location and there is not already a country code attribute. Does not update the country code
+     * if the country code attribute is already set.
+     *
+     * @param metacard
+     */
+    private void setCountryCode(Metacard metacard) {
+        Optional<String> wktLocation = Optional.ofNullable(metacard.getLocation());
+
+        if (wktLocation.isPresent() && hasCountryCode(metacard)) {
```

Comment:
I think I disagree with naming the method hasCountryCode - now it's confusing - hasCountryCode returns true when it doesn't have a country code attribute. Plus the description says it does not update the cc if the cc attribute is set, but this if statement says if the wkt is present and the metacard has a country code. It works but doesn't make sense trying to read it.